### PR TITLE
Enforce branch on test remote repository

### PR DESCRIPTION
**Pull request by jrangelramos**
# Changes

- :broom: Set branch on remote repository test.
- It also contains a fix environment [error](https://github.com/knative/func/actions/runs/11284604957/job/31403100368?pr=2540#step:9:136) on python test-template run on GitHub action

/kind cleanup

### CODE REVIEW

**Comment by jrangelramos** [comment]
This will fix errors on midstream CI (oncluster builds)

**Review by matzew — APPROVED** [review]
/lgtm
/approve

File: Makefile [hunk 1]
```diff
@@ -152,7 +152,7 @@ test-node: ## Test Node templates
 	cd templates/node/http && npm ci && npm test && rm -rf node_modules
 
 test-python: ## Test Python templates
-	cd templates/python/cloudevents && pip3 install -r requirements.txt && python3 test_func.py && rm -rf __pycache__
+	cd templates/python/cloudevents && pip3 install -r requirements.txt --break-system-packages && python3 test_func.py && rm -rf __pycache__
```
[the remark this answers is not in the document]
Comment (jrangelramos) [reply Makefile:155]: see this previous action run https://github.com/knative/func/actions/runs/11284604957/job/31403100368?pr=2540  

it is failing on test-templates for python.

[the remark this answers is not in the document]
Comment (jrangelramos) [reply Makefile:155]: I see 2 suggestion to fix this issue (on the log messages). One is to create a virtual env with `python3 -m venv path/to/venv` and other using the risky `--break-system-package` flag. Maybe we should try the first one first (?)

[the remark this answers is not in the document]
Comment (matejvasek) [reply Makefile:155]: I think using venv is pretty standard.

[the remark this answers is not in the document]
Comment (matejvasek) [reply Makefile:155]: I think you need venv anyway if you do not want to run tests under root, right?

[the remark this answers is not in the document]
Comment (matejvasek) [reply Makefile:155]: Also the knative tests gets `no such option: --break-system-packages`, I do not know why.

**Comment by matejvasek** [comment]
```
cd templates/python/cloudevents && python -m venv .venv && source .venv/bin/activate && pip3 install -r requirements.txt && python3 test_func.py && rm -rf __pycache__
/bin/sh: 1: python: not found
make: *** [Makefile:155: test-python] Error 127
```

**Comment by jrangelramos** [comment]
@matzew @matejvasek can you please re-lgtm

**Pull request merged**